### DISCUSSION TITLE
Install Pyleecan

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#345] [General]
Hello @BonneelP, 

there seems to be some issues (or uncertainities) while installing and runing pyleecan, now and then. Therefore I think the Installation Instructions on the website should be updated and detailed. Maybe you got some time to collect all the recent issues users had and we try to make a troubleshooting chapter out of it.

BR, Sebastian

**Comment by mjfwest** [comment]
I've a follow up question about installing in editable or developer mode.    It seems that installing using pip -e, as you would do if you were developing, breaks the code generation as the Class name is expanded to the full path. 
```python
File "/home/martin/repositories/pyleecan/pyleecan/Classes/LamSlotWind.py", line 436
    super(/home/martin/repositories/pyleecan/pyleecan/Generator/ClassesRef/Machine/LamSlotWind, self).__init__(slot=slot, L1=L1, mat_type=mat_type, Nrvd=Nrvd, Wrvd=Wrvd, Kf1=Kf1, is_internal=is_internal, Rint=Rint, Rext=Rext, is_stator=is_stator, axial_vent=axial_vent, notch=notch, yoke_notch=yoke_notch, bore=bore)
          ^
SyntaxError: invalid syntax
```

This may be a bug, caused by an assumption about the development environment, but also it could just be a lack of documentation about how it should be done?

**Reply by BonneelP** [reply]
Hello,

Thank you for testing pyleecan on Linux. We are mostly working on Windows (in particular due to the FEMM coupling #104) but we want to enable the use of Linux (since there are other solvers than FEMM). 

Indeed the error was linked to an environment assumption on the path format. 6ec0d87659eebd6d81f08a9cffd329234670b917 should have corrected the issue. 

Best regards,
Pierre
